tests/test_pysv.py

```diff
@@ -97,5 +101,52 @@ def incr(self, amount):
     run_tester(tester, target, simulator)
 
 
+def test_monitor(target, simulator):
+    class DelayedDUT(m.Circuit):
+        io = m.IO(A=m.In(m.Bits[4]), B=m.In(m.Bits[4]), O=m.Out(m.Bits[4]))
+        io += m.ClockIO(has_enable=True)
+        io.O @= m.Register(m.Bits[4], has_enable=True)()(dut()(io.A, io.B),
+                                                         CE=io.CE)
+
+    # TODO: Using a subclass breaks pysv
+    # class Monitor(fault.PysvMonitor):
+    class Monitor:
```

Comment:
@Kuree is there anything specific I need to do to make pysv work with generic python subclasses? When I use the abstract base class, I get an error
```
>       clear_imports(Monitor)

tests/test_pysv.py:128:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../pysv/pysv/util.py:92: in clear_imports
    func_defs = make_unique_func_defs(func_defs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

func_defs = [<class 'tests.test_pysv.test_monitor.<locals>.Monitor'>]

    def make_unique_func_defs(func_defs):
        # first remove any functions that's class method
        result = []
        classes = []
        for func_def in func_defs:
            if type(func_def) == type:
                if func_def not in classes:
                    result.append(func_def)
                    classes.append(func_def)
            else:
                if hasattr(func_def, "func_def"):
                    func_def = func_def.func_def
>               if func_def.parent_class is not None:
E               AttributeError: type object 'Monitor' has no attribute 'parent_class'

../pysv/pysv/util.py:74: AttributeError
```

Reply:
I never tried subclass so there might be some assumptions in the code that break the usage. I will do some test locally.

Reply:
It works with normal subclass but not metaclass. I will implement additional check to catch metaclass inheritance.

Reply:
Should be fixed with the latest release.